### PR TITLE
[8.1] TS: username instead of DN

### PR DESCRIPTION
**Pull request by fstagni**
- [x] add to wiki -- similar to TQDB

BEGINRELEASENOTES

*TransformationSystem
CHANGE: use userName instead of DN

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
For the update script for TQDB:

Is there an `f` missing for the query string?
```python
for ownerDN, owner in dnToOwner.items():
    query = "UPDATE `tq_TaskQueues` SET `Owner` = {owner} WHERE `OwnerDN` = {ownerDN}"
#          ^     
    res = tqDB._query(query)
    if not res["OK"]:
        print(f"ERROR -- {res['Message']}")
        exit(1)
````
https://github.com/DIRACGrid/DIRAC/wiki/DIRAC-8.1#remove-of-diracsetup-and-ownerdn

File: src/DIRAC/TransformationSystem/DB/TransformationDB.py [hunk 1]
```diff
@@ -997,7 +997,7 @@ def deleteTasks(self, transName, taskIDbottom, taskIDtop, author="", connection=
             res = self.__removeTransformationTask(transID, taskID, connection=connection)
             if not res["OK"]:
                 return res
-        message = "Deleted tasks from %d to %d" % (taskIDbottom, taskIDtop)
+        message = f"Deleted tasks from {taskIDbottom} to {taskIDtop}" % (taskIDbottom, taskIDtop)
```
Comment (andresailer) [inline src/DIRAC/TransformationSystem/DB/TransformationDB.py]:
```suggestion
        message = f"Deleted tasks from {taskIDbottom} to {taskIDtop}"
```

**Review by andresailer — CHANGES_REQUESTED** [review]
One f-string replacement isn't correct

**Comment by fstagni** [comment]
> For the update script for TQDB:
> 
> Is there an `f` missing for the query string?
> 
> ```python
> for ownerDN, owner in dnToOwner.items():
>     query = "UPDATE `tq_TaskQueues` SET `Owner` = {owner} WHERE `OwnerDN` = {ownerDN}"
> #          ^     
>     res = tqDB._query(query)
>     if not res["OK"]:
>         print(f"ERROR -- {res['Message']}")
>         exit(1)
> ```
> 
> https://github.com/DIRACGrid/DIRAC/wiki/DIRAC-8.1#remove-of-diracsetup-and-ownerdn

Thanks, fixed. A very similar script (or better a patch of this one) will be needed to update TransformationDB too.

**Comment by andresailer** [comment]
> Thanks, fixed. A very similar script (or better a patch of this one) will be needed to update TransformationDB too.

Yeah, that's why I looked at the change notes.
My biggest question would be about what to do about DNs that are no longer registered, either because the user is no longer present, or they changed their DN for some reason

**Comment by fstagni** [comment]
> > Thanks, fixed. A very similar script (or better a patch of this one) will be needed to update TransformationDB too.
> 
> My biggest question would be about what to do about DNs that are no longer registered, either because the user is no longer present, or they changed their DN for some reason

Ah, good one. Suggestions?

**Comment by andresailer** [comment]
Either use a default username, or add a map of old DN to username, or just "inactive" or something? Those should be for no-longer active transformations anyway.

**Comment by fstagni** [comment]
> Either use a default username, or add a map of old DN to username, or just "inactive" or something? Those should be for no-longer active transformations anyway.

I have updated the script. Please have a look at that, and review this PR too.

**Comment by andresailer** [comment]
https://github.com/DIRACGrid/DIRAC/wiki/DIRAC-8.1#remove-of-diracsetup-and-ownerdn
```mysql
ALTER TABLE `Transformations` MODIFY COLUMN `AuthorDN` VARCHAR(255)ssssssssssssssssssssssssssssssssssssssssssssssss;
```
?

**Comment by fstagni** [comment]
;-) fixed

**Comment by fstagni** [comment]
@andresailer review OK now? I have addressed the requested changes.

**Review by andresailer — APPROVED** [review]

**Pull request merged**